lib/posix-socket/socket_vnops.c

```diff
@@ -305,15 +305,13 @@ static int posix_socket_vfscore_poll(struct vnode *vnode, unsigned int *revents,
 	sock = (struct posix_socket_file *)vnode->v_data;
 
 	ret = posix_socket_poll(sock, revents, ecb);
-	if (ret < 0)
-		return -ret;
 	if (unlikely(ret < 0)) {
 		PSOCKET_ERR("poll on socket %d failed: %d\n",
 			    sock->vfs_file->fd, (int)ret);
 		ret = -ret;
 	}
 
-	return 0;
+	return ret;
```

Comment:
Same as for `ioctl`. In addition, returning a positive number here also does not bring any benefit because currently any non-zero value is interpreted as error condition in `eventpoll.c`.

Reply:
I think I marked this as resolved a bit too fast. There is still a return in line 311 missing, right? Currently, this ignores all errors.

Reply:
Yes, for sure. Same bug as in ioctl 😃